### PR TITLE
Feat/improve doc

### PR DESCRIPTION
**Pull request by hrzn**
Improve documentation

* Remove abstract base classes from documentation
* Copied parameters docstrings from parents classes where needed (torch models)
* Improve Torch models docuemntation
* Create pages for Kalman and Gaussian processes
* Various small improvements here and there

I re-opened the PR because the tests didn't seem to start on #423

### CODE REVIEW

File: darts/models/gaussian_process_filter.py [hunk 1]
```diff
@@ -1,3 +1,8 @@
+"""
+Gaussian Processes
+------------------
```
Comment (pennfranc) [inline darts/models/gaussian_process_filter.py:3]: In some cases you removed these lines. I don't have a preference but I suppose we should be consistent

File: darts/models/regression_model.py [hunk 2]
```diff
@@ -3,9 +3,11 @@
 ----------------
 
 A `RegressionModel` forecasts future values of a target series based on lagged values of the target values
-and possibly lags of an exoneous series. They can wrap around any regression model having a `fit()`
-and `predict()` functions (e.g. scikit-learn regression models), and are using
-`sklearn.linear_model.LinearRegression` by default.
+and possibly lags of a covariate series. They can wrap around any regression model having a `fit()`
```
Comment (pennfranc) [inline darts/models/regression_model.py:6]:
```suggestion
and possibly lags of a covariate series. They can wrap around any regression model having `fit()`
```

File: darts/models/torch_forecasting_model.py [hunk 3]
```diff
@@ -610,29 +628,10 @@ def predict_from_dataset(self,
...
-        All predictions are produced in a recurrent way by taking as input
-        - the previous target value, which will be set to the last known target value for the first prediction,
-          and for all other predictions it will be set to the previous prediction
-        - the previous hidden state
-        - the current covariates (if the model was trained with covariates)
-        As a result, if covariates were used, `n` covariates have to be available into the future.
-
-        If some series in the `input_series_dataset` have more time steps than `input_chunk_length`,
-        only the last `input_chunk_length` time steps will be considered.
+        This method allows for training with a specific `darts.utils.data.InferenceDataset` instance. These datasets
```
Comment (pennfranc) [inline darts/models/torch_forecasting_model.py:631]:
```suggestion
        This method allows for predicting with a specific `darts.utils.data.InferenceDataset` instance. These datasets
```

File: darts/models/torch_forecasting_model.py [hunk 4]
```diff
@@ -610,29 +628,10 @@ def predict_from_dataset(self,
...
+        implement a PyTorch `Dataset`, and specify how the target and covariates are sliced for inference.
+        In most cases, you'll rather want to call `fit()` instead, which will create an appropriate `InferenceDataset`
```
Comment (pennfranc) [inline darts/models/torch_forecasting_model.py:633]:
```suggestion
        In most cases, you'll rather want to call `predict()` instead, which will create an appropriate `InferenceDataset`
```

**Review by pennfranc — APPROVED** [review]
I think I found a couple of typos and inconsistencies, but other than that it looks really nice, thanks a lot!

File: darts/models/gaussian_process_filter.py (hunk 1 above)
Comment (hrzn) [reply darts/models/gaussian_process_filter.py:3]: Yes, removing the dotted lines removes the page from the generated doc.

**Pull request merged**